### PR TITLE
feat(shares:events) emit events when share links are created/updated

### PR DESCRIPTION
**Pull request by kschelonka**
Emit events when share links are created/updated.

[POCKET-10079]

[POCKET-10079]: https://mozilla-hub.atlassian.net/browse/POCKET-10079?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infrastructure/pocket-event-bridge/src/event-rules/shares-api-events/eventConfig.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const eventConfig = {
```
Comment (kschelonka) [inline infrastructure/pocket-event-bridge/src/event-rules/shares-api-events/eventConfig.ts:1]: infrastructure source of truth for rules

File: servers/shared-snowplow-consumer/src/snowplow/EventHandler.ts [hunk 2]
```diff
@@ -13,8 +9,7 @@ export class EventHandler {
   }
 
   public trackObjectUpdate<T = any>(
-    tracker: Tracker,
-    objectUpdate: ObjectUpdate & CommonEventProperties<T>,
+    ...[tracker, objectUpdate]: Parameters<typeof baseTrackObjectUpdate<T>>
```
Comment (kschelonka) [inline servers/shared-snowplow-consumer/src/snowplow/EventHandler.ts:12]: use conditional parameters and propagate the generics to get proper argument types without editing the generated file (CommonEventProperties was not exported)

File: servers/shares-api/src/config/index.ts [hunk 3]
```diff
@@ -22,6 +22,11 @@ export const config = {
     region: process.env.AWS_REGION || 'us-east-1',
     endpoint: localAwsEndpoint,
     maxBackoff: 3000, // in ms, max amount of backoff time allowed for multiple requests
+    eventBus: {
+      name:
+        process.env.EVENT_BUS_NAME || 'PocketEventBridge-Dev-Shared-Event-Bus',
+      source: 'shares-api-events',
```
Comment (kschelonka) [inline servers/shares-api/src/config/index.ts:28]: must match config source of truth (refactor later to have shared schema package)

File: servers/shares-api/src/events/EventBus.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import {
+  EventBridgeClient,
+  PutEventsCommand,
+} from '@aws-sdk/client-eventbridge';
+import { PocketShare } from '../__generated__/types';
+import * as Sentry from '@sentry/node';
+import { serverLogger } from '@pocket-tools/ts-logger';
+import { config } from '../config';
+
+type PocketShareEvent = {
```
Comment (kschelonka) [inline servers/shares-api/src/events/EventBus.ts:10]: must match snowplow event schema (refactor later to have shared schema package)

File: servers/shares-api/src/events/EventBus.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  target_url: string;
+  created_at: number; // epoch time in seconds
+  slug: string;
+  note_length: number;
+  quote_count: number;
+};
+
+enum PocketShareEventType {
```
Comment (kschelonka) [inline servers/shares-api/src/events/EventBus.ts:18]: must match detail-types in event bridge source of truth

File: servers/shared-snowplow-consumer/src/config/index.ts [hunk 6]
```diff
@@ -64,6 +64,10 @@ export const config = {
       userApi: 'pocket-user-api',
       collectionApi: 'pocket-collection-api',
       shareableListsApi: 'pocket-shareable-lists-api',
+      // todo: kat - confused about this; why not use event bridge
+      // source or update the source field?
```
Comment (bassrock) [inline servers/shared-snowplow-consumer/src/config/index.ts:68]: ✔️

File: servers/shared-snowplow-consumer/src/snowplow/EventHandler.ts (hunk 2 above)
Comment (bassrock) [reply servers/shared-snowplow-consumer/src/snowplow/EventHandler.ts:12]: smart. this is much better then what i did before

**Review by bassrock — APPROVED** [review]

**Pull request merged**